crates/leon/src/error.rs

```diff
@@ -0,0 +1,131 @@
+#[cfg(feature = "miette")]
+use miette::{Diagnostic, SourceSpan};
+use thiserror::Error;
+
+#[cfg_attr(feature = "miette", derive(Diagnostic))]
+#[derive(Debug, Error)]
+pub enum RenderError {
+    /// A key was missing from the provided values.
+    #[error("missing key `{0}`")]
+    MissingKey(String),
+
+    /// An I/O error passed through from [`Template::render_into`].
+    #[error(transparent)]
+    Io(#[from] std::io::Error),
+}
+
+/// An error that can occur when parsing a template.
+///
+/// This is a rich miette-powered error which will highlight the source of the
+/// error in the template when output (with miette's `fancy` feature). If you
+/// don't want or need that, you can disable the `miette` feature and a simpler
+/// opaque error will be substituted.
+#[cfg(feature = "miette")]
+#[derive(Clone, Debug, Error, PartialEq, Eq)]
+#[error(transparent)]
+pub struct ParseError(pub Box<InnerParseError>);
```

Comment:
I think you could use `#[derive(Diagnostic)]` and then put `#[diagnostic(transparent)]` on the field to forward `Diagnostic` implementation.